caluma/caluma_form/models.py

```diff
@@ -295,11 +295,13 @@ def copy(self, family=None, user=None):
 
         return new_document
 
+    def is_root(self):
+        return self == self.family
+
     @cached_property
     def last_modified_answer(self):
-        return (
-            Answer.objects.filter(document__family=self).order_by("modified_at").last()
-        )
+        _filter = "document__family" if self.is_root() else "document"
```

Comment:
It took me quite a while to understand why this is required. I think a comment and an extended commit message would be very useful.

Somethings along the lines of "for row documents, we want the most recently changed answer within this document, not traversing to the parent/root document. Only for root documents should we look at all answers, including rows"

Reply:
You are right, added comment and commit message.b